jetty-http/src/main/java/org/eclipse/jetty/http/HttpCookie.java

```diff
@@ -92,6 +114,8 @@ public HttpCookie(String setCookie)
         _comment = cookie.getComment();
         _version = cookie.getVersion();
         _expiration = _maxAge < 0 ? -1 : System.nanoTime() + TimeUnit.SECONDS.toNanos(_maxAge);
+        // TODO support for SameSite values has not yet been added to java.net.HttpCookie
+        _sameSite = SameSite.EXCLUDED;
```

Comment:
See the code we have in 

https://github.com/eclipse/jetty.project/blob/jetty-9.4.20.v20190809/jetty-server/src/main/java/org/eclipse/jetty/server/Response.java#L231-L262

We work around javax.servlet.http.Cookie not having `httponly` by allowing it to be encoded in the comment.  So you should do the same there for `SameSite`, and perhaps here as well?

Reply:
Hmm, this complicates things a bit. There is also a bit of inconsistency here in that `HttpCookie(String setCookie)` doesn't check for the `__HTTP_ONLY__` flag in the comment so if I update this it may make sense to check for the HttpOnly flag in the comment too.

I started on some refactoring, but it seems better to get your feedback first. Which change would you prefer?
1. Add a check in the `Response.java` that checks the comment for a SameSite flag after the current HttpOnly check. Leave the `_sameSite=null` here to avoid repeating code.
2. Add a new `public HttpCookie(java.net.HttpCookie)` constructor that initializes a new `org.eclipse.jetty.http.HttpCookie` from an existing `java.net.HttpCookie`
 and checks the comments for setting cookie parameters. This would remove the need to check the comment in `Response.java`.
3. Pull the comment parsing out of `Response.java` into a private method of `org.eclipse.jetty.http.HttpCookie` called something like `parseCommentForFlags()` that is called at the end of constructors to set fields based on the comment (this would remove the need to check the comment in Response too).

(or something else)

I was leaning towards option 3, but it means a bit of test refactoring (which I'm happy to do but didn't want to commit to before checking).

Reply:
I'm kind of OK with 3.... but perhaps instead have static methods on HttpCookie like:
```
   HttpCookie isHttpOnlyInComment(String comment)
   SameSite getSameSiteFromComment(String comment)
   String getCommentWithoutFlags(String comment)
```
and then use them when calling the constructor?

Reply:
Thanks for the feedback. I've added your suggestion in the most recent commit (f482e40ffcb862e82434f27879819c00afb7e4dc), but with `boolean isHttpOnlyInComment(String comment)` rather than `HttpCookie isHttpOnlyInComment(String comment)` as I wasn't sure how it would make sense to return an HttpCookie from this method.